File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDOGpsControlParamModel.kt
//
//  IDOGpsControlParamModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

///
/// Control GPS event number

open class IDOGpsControlParamModel(operate: Int, type: Int) : IDOBaseModel {

    /// 1: Control 
    /// 2: Query                                
    @SerializedName("operate")
    var operate: Int = operate

    /// 1: Enable log
    /// 2: Disable log 
    /// 3: Write AGPS data 
    /// 4: Erase AGPS data 
    /// 5: Write GPS firmware 
    @SerializedName("type")
    var type: Int = type


    

    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}
    

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDOGpsControlParamModel.swift
//
//  IDOGpsControlParamModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/12.
//

import Foundation
/// Control GPS event number
@objcMembers
public class IDOGpsControlParamModel: NSObject, IDOBaseModel {
    /// 1: Control 
    /// 2: Query                                
    public var operate: Int
    /// 1: Enable log 
    /// 2: Disable log 
    /// 3: Write AGPS data 
    /// 4: Erase AGPS data 
    /// 5: Write GPS firmware 
    public var type: Int
    
    enum CodingKeys: String, CodingKey {
        case operate = "operate"
        case type = "type"
    }
    
    public init(operate: Int,type: Int) {
        self.operate = operate
        self.type = type
    }
    
    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}

